src/impl/websocket.cpp

```diff
@@ -48,7 +49,8 @@ void WebSocket::open(const string &url) {
 		throw std::logic_error("WebSocket must be closed before opening");
 
 	if (config.proxyServer) {
-		PLOG_WARNING << "Proxy server support for WebSocket is not implemented";
+		mIsProxied = true;
+		mProxy = config.proxyServer;
```

Comment:
You should not need to copy the field as `config` is actually a member. Also, you might use directly `config.proxyServer` instead of introducing the `mIsProxied` state.